File: app/src/main/java/com/sharpdroid/registroelettronico/Adapters/MarkAdapter.java
package com.sharpdroid.registroelettronico.Adapters;

import android.content.Context;
import android.graphics.drawable.ColorDrawable;
import android.support.v4.content.ContextCompat;
import android.support.v7.widget.RecyclerView;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.sharpdroid.registroelettronico.Databases.Entities.Grade;
import com.sharpdroid.registroelettronico.R;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

import butterknife.BindView;
import butterknife.ButterKnife;
import de.hdodenhof.circleimageview.CircleImageView;

import static com.sharpdroid.registroelettronico.Utils.Metodi.getMarkColor;
import static com.sharpdroid.registroelettronico.Utils.Metodi.sortMarksByDate;

public class MarkAdapter extends RecyclerView.Adapter<MarkAdapter.MarkHolder> {
    float target;
    private Context mContext;
    private SimpleDateFormat format = new SimpleDateFormat("d MMMM yyyy", Locale.ITALIAN);
    private List<Grade> CVDataList;

    public MarkAdapter(Context mContext) {
        this.mContext = mContext;
        CVDataList = new ArrayList<>();
    }

    public void addAll(List<Grade> list) {
        CVDataList = sortMarksByDate(list);
        notifyDataSetChanged();
    }

    public void clear() {
        CVDataList.clear();
        notifyDataSetChanged();
    }

    public void setTarget(float t) {
        target = t;
        notifyDataSetChanged();
    }

    @Override
    public MarkHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new MarkHolder(LayoutInflater.from(mContext).inflate(R.layout.adapter_mark, parent, false));
    }

    @Override
    public void onBindViewHolder(MarkHolder holder, int position) {

        Grade mark = CVDataList.get(position);
        holder.color.setImageDrawable(new ColorDrawable(ContextCompat.getColor(mContext, getMarkColor(mark.getMValue(), target))));
        holder.mark.setText(mark.getMStringValue());

        holder.content.setText(mark.getMNotes().trim());

        holder.content.setVisibility((TextUtils.isEmpty(holder.content.getText()) ? View.GONE : View.VISIBLE));

        holder.type.setText(mark.getMType());
        holder.date.setText(format.format(mark.getMDate()));
    }

    @Override
    public int getItemCount() {
        return CVDataList.size();
    }


    class MarkHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.color)
        CircleImageView color;
        @BindView(R.id.mark)
        TextView mark;
        @BindView(R.id.content)
        TextView content;
        @BindView(R.id.date)
        TextView date;
        @BindView(R.id.type)
        TextView type;

        MarkHolder(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }
    }


}

File: app/src/main/java/com/sharpdroid/registroelettronico/adapters/MarkAdapter.kt
package com.sharpdroid.registroelettronico.adapters

import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.support.v4.content.ContextCompat
import android.support.v7.widget.RecyclerView
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.sharpdroid.registroelettronico.R
import com.sharpdroid.registroelettronico.database.entities.Grade
import com.sharpdroid.registroelettronico.utils.Metodi.getMarkColor
import com.sharpdroid.registroelettronico.utils.Metodi.sortMarksByDate
import de.hdodenhof.circleimageview.CircleImageView
import kotlinx.android.synthetic.main.adapter_mark.view.*
import java.text.SimpleDateFormat
import java.util.*

class MarkAdapter(private val mContext: Context) : RecyclerView.Adapter<MarkAdapter.MarkHolder>() {
    internal var target: Float = 0f
    private val format = SimpleDateFormat("d MMMM yyyy", Locale.ITALIAN)
    private val data = mutableListOf<Grade>()

    fun addAll(list: List<Grade>) {
        data.addAll(sortMarksByDate(list))
        notifyDataSetChanged()
    }

    fun clear() {
        data.clear()
        notifyDataSetChanged()
    }

    fun setTarget(t: Float) {
        target = t
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
            MarkHolder(LayoutInflater.from(mContext).inflate(R.layout.adapter_mark, parent, false))

    override fun onBindViewHolder(holder: MarkHolder, position: Int) {

        val mark = data[position]
        holder.color.setImageDrawable(ColorDrawable(ContextCompat.getColor(mContext, getMarkColor(mark.mValue, target))))
        holder.mark.text = mark.mStringValue

        holder.content.text = mark.mNotes.trim { it <= ' ' }

        holder.content.visibility = if (TextUtils.isEmpty(holder.content.text)) View.GONE else View.VISIBLE

        holder.type.text = mark.mType
        holder.date.text = format.format(mark.mDate)
    }

    override fun getItemCount() = data.size

    inner class MarkHolder(l: View) : RecyclerView.ViewHolder(l) {
        var color: CircleImageView = l.color
        var mark: TextView = l.mark
        var content: TextView = l.content
        var date: TextView = l.date
        var type: TextView = l.type
    }
}
